src/Dfe.PlanTech.Web/ViewComponents/CategorySectionViewComponent.cs

```diff
@@ -21,90 +27,96 @@ public async Task<IViewComponentResult> InvokeAsync(Category category)
+            TotalSectionCount = category.Sections.Count,
+            CategorySectionDto = await GetCategorySectionDto(category).ToListAsync(),
+            ProgressRetrievalErrorMessage = category.RetrievalError
+                ? "Unable to retrieve progress, please refresh your browser."
+                : null
         };
     }
 
-    private void LogErrorWithUserFeedback(Section categorySection, CategorySectionDto categorySectionDto)
-    {
-        categorySectionDto.Slug = null;
-        _logger.LogError("No Slug found for Subtopic with ID: {categorySectionId}", categorySection.Sys.Id);
-        categorySectionDto.NoSlugForSubtopicErrorMessage = string.Format("{0} unavailable", categorySection.Name);
-    }
-
-    private static void SetCategorySectionDtoTagWithRetrievalError(CategorySectionDto categorySectionDto)
-    {
-        categorySectionDto.TagColour = TagColour.Red.ToString();
-        categorySectionDto.TagText = "UNABLE TO RETRIEVE STATUS";
-    }
-
-    private static void SetCategorySectionDtoTagWithCurrentStatus(Category category, Section categorySection, CategorySectionDto categorySectionDto)
-    {
-        var sectionStatusCompleted = category.SectionStatuses.FirstOrDefault(sectionStatus => sectionStatus.SectionId == categorySection.Sys.Id)?.Completed;
-
-        if (sectionStatusCompleted != null)
-        {
-            categorySectionDto.TagColour = sectionStatusCompleted == 1 ? TagColour.Blue : TagColour.LightBlue;
-            categorySectionDto.TagText = sectionStatusCompleted == 1 ? "COMPLETE" : "IN PROGRESS";
-        }
-        else
-        {
-            categorySectionDto.TagColour = TagColour.Grey.ToString();
-            categorySectionDto.TagText = "NOT STARTED";
-        }
-    }
-
-    private IEnumerable<CategorySectionDto> GetCategorySectionViewComponentViewModel(Category category)
+    private async IAsyncEnumerable<CategorySectionDto> GetCategorySectionDto(Category category)
     {
         foreach (var section in category.Sections)
         {
-            var categorySectionDto = new CategorySectionDto()
-            {
-                Slug = section.InterstitialPage?.Slug,
-                Name = section.Name
-            };
-
-            if (string.IsNullOrWhiteSpace(categorySectionDto.Slug)) LogErrorWithUserFeedback(section, categorySectionDto);
-            else if (category.RetrievalError) SetCategorySectionDtoTagWithRetrievalError(categorySectionDto);
-            else SetCategorySectionDtoTagWithCurrentStatus(category, section, categorySectionDto);
-
-            yield return categorySectionDto;
+            var sectionStatus = category.SectionStatuses.FirstOrDefault(sectionStatus => sectionStatus.SectionId == section.Sys.Id);
+
+            if (string.IsNullOrWhiteSpace(section.InterstitialPage.Slug))
+                _logger.LogError($"No Slug found for Subtopic with ID: ${section.Sys.Id}/ name: {section.Name}");
```

Comment:
1. Don't use string interpolation for logging messages (tl;dr because the log message might not actually get sent due to log level, you're potentially executing code + creating objects in memory unnecessarily)
2. ... Did I write this code? because the dollar sign in `${section.Sys.Id}` is unnecessary, but I accidentally do that all the time when I've been programming JS/TS and come back to C#....

Reply:
I could just as likely have done that post some JS or ruby 😆 fixed, thanks, and fixed the logging messages